### DISCUSSION TITLE
Any plans to implement LazyCollection?

### DISCUSSION THREAD

**Discussion by jiaoliness** [Power-Components/livewire-powergrid#638] [Ideas]
Loading a very large dataset into the table takes quite a bit of time. The SQL query itself is very fast, but loading all the individual results into memory when each result is transformed into a Collection object takes time, for me around 10 seconds for pageload for 200k entries.

**Comment by luanfreitasdev** [comment]
Hi, we know that in the database it will only bring up using pagination, and that shouldn't be slow if you specify a very high `perPage`.

Since this table has a lot of records, I recommend using "**min**" mode in [Footer::showRecordCount](https://livewire-powergrid.com/#/table/features-setup?id=showrecordcount). This will eliminate the total table count.
```php
Footer::make()
      ->showPerPage()
      ->showRecordCount('min'),
```

![image](https://user-images.githubusercontent.com/33601626/192516351-e2b2eb61-4fe8-435a-b35a-cbcf33dd7422.png)

Make sure you don't use Collection in the datasource method;
```php
public function datasource(): Builder
// ...
```

You can also specify which columns the Builder should query.
```php
public function datasource(): Builder
{
     return User::query()->select('id', 'name');
}
```

Make sure you are using 'Eager Loaging' when retrieving data from a relationship.
```php
public function datasource(): Builder
{
     return Dish::with(['category:id,name', 'kitchen']);
}
```

Tell us if it worked.
